receiver/prometheusreceiver/factory.go

```diff
@@ -65,7 +66,10 @@ func CustomUnmarshalerFunc(v *viper.Viper, viperKey string, intoCfg interface{})
 	// YAML unmarshaling routines so we need to do it explicitly.
 
 	// Unmarshal our config values (using viper's mapstructure)
-	err := v.UnmarshalKey(viperKey, intoCfg)
+	errorOnUnused := func(decoderCfg *mapstructure.DecoderConfig) {
+		decoderCfg.ErrorUnused = true
```

Comment:
What does ErrorUnused flag do?

Reply:
It errors if there is a key in the config that is not used, I will add a comment to the code.